pinot-spi/src/main/java/org/apache/pinot/spi/utils/builder/ControllerRequestURLBuilder.java

```diff
@@ -379,26 +379,45 @@ public String forDeleteTableWithType(String tableName, String tableType) {
   }
 
   public String forSegmentListAPI(String tableName) {
-    return forSegmentListAPI(tableName, null, false);
+    return forSegmentListAPI(tableName, null, false, Long.MIN_VALUE, Long.MAX_VALUE, false);
   }
 
   public String forSegmentListAPI(String tableName, String tableType) {
-    return forSegmentListAPI(tableName, tableType, false);
+    return forSegmentListAPI(tableName, tableType, false, Long.MIN_VALUE, Long.MAX_VALUE, false);
   }
 
-  public String forSegmentListAPI(String tableName, @Nullable String tableType, boolean excludeReplacedSegments) {
+  public String forSegmentListAPI(String tableName, String tableType, boolean excludeReplacedSegments) {
+    return forSegmentListAPI(tableName, tableType, excludeReplacedSegments, Long.MIN_VALUE, Long.MAX_VALUE, false);
+  }
+
+  public String forSegmentListAPI(String tableName, @Nullable String tableType, boolean excludeReplacedSegments,
+      long startTimestamp, long endTimestamp, boolean excludeOverlapping) {
     String url = StringUtil.join("/", _baseUrl, "segments", tableName);
+    String urlParameter = "";
```

Comment:
nit: replace it with `StringBuilder` approach?

Reply:
fixed